src/task_monitor.rs

```diff
@@ -126,55 +72,15 @@ pub struct TaskMonitor {
     /// when shutdown is called we want to be able to gracefully
     /// await the join handles - which requires ownership of the handle and by
     /// extension the instance.
-    instance:                  RwLock<Option<RunningInstance>>,
-    database:                  Arc<Database>,
-    identity_manager:          SharedIdentityManager,
-    tree_state:                TreeState,
-    batch_insert_timeout_secs: u64,
-
-    // Finalization params
-    scanning_window_size:           u64,
-    scanning_chain_head_offset:     u64,
-    time_between_scans:             Duration,
-    max_epoch_duration:             Duration,
-    // TODO: docs
-    batch_deletion_timeout_seconds: i64,
-    // TODO: docs
-    min_batch_deletion_size:        usize,
-    monitored_txs_capacity:         usize,
+    instance: RwLock<Option<RunningInstance>>,
```

Comment:
Consider moving `handles` and `shutdown_sender` directly into `TaskMonitor`, removing `RunningInstance`.

Reply:
Eventually I want to get rid of TaskMonitor entirely - or modify it significantly. But one thing at a time